### PR TITLE
Translate: Strict mode

### PR DESCRIPTION
**Pull request by d4vsanchez**
I've done translating `Strict mode` to Spanish. #4

I'm having some doubts with some stuff in the translation and I hope you can help me with them if you have a better way to translate it:

* **commit phase**: I'm translating this as `fase de confirmación`.
* **refs**: This article talks about refs in multiple ways: `string ref`, `callback ref`, `object ref`. I've basically left them as is; I can instead translate them `referencia de string` (to comply with the translation guideline table), `referencia de objeto` and `referencia de callback`. What do you think it's better?

cc: @alejandronanez @dmoralesm @carburo

### CODE REVIEW

File: content/docs/addons-test-utils.md [hunk 1]
```diff
@@ -44,13 +44,13 @@ var ReactTestUtils = require('react-dom/test-utils'); // ES5 with npm
 
 ### `act()`
 
-To prepare a component for assertions, wrap the code rendering it and performing updates inside an `act()` call. This makes your test run closer to how React works in the browser.
+Para preparar un componente en ser asegurado, debes envolver el código que lo renderiza y que realiza actualizaciones sobre este en un llamado a `act()`. Esto hace que tus pruebas corran de una forma más parecida a como lo hace React en el navegador.
```
Comment (jonalvarezz) [inline content/docs/addons-test-utils.md]: Qué tal: "Para preparar la asertividad en un componente..."

File: content/docs/addons-test-utils.md [hunk 2]
```diff
@@ -122,7 +122,7 @@ it('can render and update a counter', () => {
 });
 ```
 
-Don't forget that dispatching DOM events only works when the DOM container is added to the `document`. You can use a helper like [`react-testing-library`](https://github.com/kentcdodds/react-testing-library) to reduce the boilerplate code.
+No olvides que la ejecución de eventos del DOM sólo funciona cuando el contenedor del DOM es agregado al `document`. Puedes utilizar una ayuda como [`react-testing-library`](https://github.com/kentcdodds/react-testing-library) para reducir todo el código repetitivo.
```
Comment (jonalvarezz) [inline content/docs/addons-test-utils.md]: Una traducción más precisa para "You can use a helper like..." sería "Puedes usar un ayudante como..."

File: content/docs/addons-test-utils.md [hunk 3]
```diff
@@ -320,19 +320,19 @@ Simulate.{eventName}(
 )
 ```
 
-Simulate an event dispatch on a DOM node with optional `eventData` event data.
+Simula la ejecución de un evento en un nodo del DOM con los datos de evento `eventData` opcionales.
 
-`Simulate` has a method for [every event that React understands](/docs/events.html#supported-events).
+`Simulate` tiene un método para [cada uno de los eventos que React comprende](/docs/events.html#supported-events).
```
Comment (jonalvarezz) [inline content/docs/addons-test-utils.md]: El "comprende" también es sinónimo de abarcar y me parece ambiguo acá. 

"Simulate` tiene un método para [cada uno de los eventos que React *entiende*" Me parecería más preciso

File: content/docs/addons-test-utils.md [hunk 4]
```diff
@@ -320,19 +320,19 @@ Simulate.{eventName}(
...
+**Haciendo clic en un elemento**
 
 ```javascript
 // <button ref={(node) => this.button = node}>...</button>
 const node = this.button;
 ReactTestUtils.Simulate.click(node);
 ```
 
-**Changing the value of an input field and then pressing ENTER.**
+**Cambiar el valor en un campo de entrada y presionar ENTER.**
```
Comment (jonalvarezz) [inline content/docs/addons-test-utils.md]: *Cambiando* el valor en un campo de entrada y *presionando*

File: content/docs/addons-test-utils.md [hunk 5]
```diff
@@ -342,8 +342,7 @@ ReactTestUtils.Simulate.change(node);
 ReactTestUtils.Simulate.keyDown(node, {key: "Enter", keyCode: 13, which: 13});
 ```
 
-> Note
->
-> You will have to provide any event property that you're using in your component (e.g. keyCode, which, etc...) as React is not creating any of these for you.
+> Nota
```
Comment (jonalvarezz) [inline content/docs/addons-test-utils.md]: Acá se borró un salto de linea, hace que Markdown pierda el formato.

File: content/docs/addons-test-utils.md [hunk 6]
```diff
@@ -342,8 +342,7 @@ ReactTestUtils.Simulate.change(node);
...
+> Se debe proveer cualquiera de las propiedades del evento que se esté usando en tu componente (p.e. keyCode, which, etc...) ya que React no creará ninguna de estas por ti.
```
Comment (jonalvarezz) [inline content/docs/addons-test-utils.md]: "*Tendrás* que proveer cualquiera de las propiedades del evento que *estés* usando en tu componente (p.e. keyCode, which, etc...) ya que React no creará ninguna de estas por ti."

**Comment by jonalvarezz** [comment]
Great job David 😁! I've added some comments.

**Comment by d4vsanchez** [comment]
Thanks Jonathan! I just made a commit applying those comments.

File: content/docs/strict-mode.md [hunk 7]
```diff
@@ -1,65 +1,65 @@
...
-When strict mode is enabled, React compiles a list of all class components using the unsafe lifecycles, and logs a warning message with information about these components, like so:
+Cuando el modo estricto está habilitado, React reune en una lista todos los componentes de clases que están usando ciclos de vida inseguros, y registra por medio de un mensaje de advertencia la información sobre estos componentes, de esta forma:
 
 ![](../images/blog/strict-mode-unsafe-lifecycles-warning.png)
 
-Addressing the issues identified by strict mode _now_ will make it easier for you to take advantage of async rendering in future releases of React.
+Solucionar los problemas identificados por el modo estricto _ahora_, hará que sea más fácil para ti aprovechar el renderizamiento asíncrono en futuras versiones de React.
 
-### Warning about legacy string ref API usage {#warning-about-legacy-string-ref-api-usage}
+### Advertir sobre el uso de la API legado de string ref {#warning-about-legacy-string-ref-api-usage}
```
Comment (alejandronanez) [inline content/docs/strict-mode.md]: I think `Advertencia` works better here! Thoughts?
```suggestion
### Advertencia sobre el uso de la API legado de string ref {#warning-about-legacy-string-ref-api-usage}
```

File: content/docs/strict-mode.md [hunk 8]
```diff
@@ -1,65 +1,65 @@
...
+> Los callback refs seguirán siendo soportados en conjunto con la nueva API de `createRef`.
 >
-> You don't need to replace callback refs in your components. They are slightly more flexible, so they will remain as an advanced feature.
+> No necesitas reemplazar los callback refs en tu componente. Son un poco más flexibles, por lo que continuarán estando como una característica avanzada.
 
-[Learn more about the new `createRef` API here.](/docs/refs-and-the-dom.html)
+[Aprende más sobre la API `createRef` aquí.](/docs/refs-and-the-dom.html)
 
-### Warning about deprecated findDOMNode usage {#warning-about-deprecated-finddomnode-usage}
+### Advertir sobre el uso del método obsoleto findDOMNode {#warning-about-deprecated-finddomnode-usage}
```
Comment (alejandronanez) [inline content/docs/strict-mode.md]: Same story here :) - I think `Advertencia` works better :)
```suggestion
### Advertencia sobre el uso del método obsoleto findDOMNode {#warning-about-deprecated-finddomnode-usage}
```

File: content/docs/strict-mode.md [hunk 9]
```diff
@@ -73,52 +73,52 @@ class MyComponent extends React.Component {
...
+El ciclo de vida de la fase de renderizado incluye los siguientes métodos de los componentes en clases:
 * `constructor`
 * `componentWillMount`
 * `componentWillReceiveProps`
 * `componentWillUpdate`
 * `getDerivedStateFromProps`
 * `shouldComponentUpdate`
 * `render`
-* `setState` updater functions (the first argument)
+* Funciones de actualización de `setState` (el primer argumento)
```
Comment (alejandronanez) [inline content/docs/strict-mode.md]: 👍

Comment (d4vsanchez) [reply content/docs/strict-mode.md]: I don't understand this one, is it OK? haha

**Comment by d4vsanchez** [comment]
Hey @alejandronanez, they were very valuable! I have made a new commit applying those suggestions and also modifying the "Advertencia" in the bullet points section..

File: content/docs/strict-mode.md (hunk 9 above)
Comment (alejandronanez) [reply content/docs/strict-mode.md]: haha it sounds good I think 😂

**Pull request merged**